sspl_test/coverage/coverage_setup.py

```diff
@@ -0,0 +1,172 @@
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import shutil
+import os
+import pwd
+import signal
+import psutil
+import time
+import sys
+from cortx.utils.process import SimpleProcess
+# Add the top level directory to the sys.path to access classes
+topdir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+os.sys.path.insert(0, topdir)
+from framework.base.sspl_constants import DATA_PATH
+
+file_path = "/opt/seagate/cortx/sspl/low-level"
+report_path = f"{DATA_PATH}coverage/sspl_xml_coverage_report.xml"
+
+PATCH_1 = """\
+from coverage import Coverage
+co = Coverage(
+        data_file='%scoverage/.sspl_coverage_report',
+        include="/opt/seagate/*",
+        omit=['*/.local/*', '*/usr/*'],
+        # config_file='%scoverage/.coveragerc',
+        )\
+"""%(DATA_PATH, DATA_PATH)
+
+PATCH_2 = """\
+        logger.info("Starting coverage report scope")
+        co.start()\
+"""
+
+PATCH_3 = """\
+def generate_cov_report(signal_number, frame):
+    logger.info('Ending Coverage Scope')
+    co.stop()
+    logger.info('coverage object stopped.')
+    co.save()
+    logger.info('coverage info saved.')
+    cov_per = co.xml_report(ignore_errors=True,
+        outfile='%scoverage/sspl_xml_coverage_report.xml')
+    logger.info(f'XML coverage report generated with coverage of {cov_per} percentage.')
+    ## Enable below code to inable HTML report generation
+ 	# html_cov_per = co.html_report(
+ 	#                     directory='%scoverage/sspl_html_coverage',
+ 	#                     ignore_errors=True,
+ 	#                 )
+ 	# logger.info(f'HTML coverage report geverated with coverage of {html_cov_per} percentage.')\
+"""%(DATA_PATH, DATA_PATH)
+
+PATCH_4 = """\
+    signal.signal(signal.SIGUSR1, generate_cov_report)\
+"""
+
+def coverage_setup():
+    """Installs pip3 coverage package. Injects different patches code from
+       coverage_code file to sspl_ll_d file, also creates target directory
+       for code coverage report and assigns permission to the directory.
+    """
+    print("Installing coverage.py")
+    _, err, return_code = SimpleProcess('python3 -m pip install coverage').run()
+    if return_code:
+        print(err)
+        return return_code
+
+    print("Creating required files for coverage..")
+    patch1_name = "initialize coverage obj for code coverage report generation"
+    patch2_name = "start the code coverage scope"
+    patch3_name = "stop coverage, save and generate code coverage report"
+    patch4_name = "signal handler for SIGUSR1 to generate code coverage report"
+
```

Comment:
How come we are not stopping the service first before proceeding here? And start after the file edit then. Same applies when coverage code is cleaned up

Reply:
addressed.